src/context-providers/builder/provider.tsx

```diff
@@ -60,6 +60,7 @@ export const builderReducer = (
             break;
         }
         case "remove-focused-element": {
+            state.focusedElement.isDirty = false;
```

Comment:
why do we need to change this before changing focus?

Reply:
I realised there is no need for this since we are already setting `focusedElement` to `null`. However I did notice that if we dirty a form and then discard changes and click on a new element, on first render of `ElementEditor`, form would still be dirty since we did not reset it when discarding changes. It is set to `false` in the `useEffect` hook on 2nd render in `SidePanel`, but there will be a quick flicker in the side panel from dirty to clean. To prevent that, I will add a `reset` when we discard changes.

Reply:
I didnt observe this behaviour though, the `isDirty` state is only present in the `focusedElement` so on switching the focused element, wont that state be lost hence become falsy?

Reply:
In line 38 of `ElementEditor`,

`updateFocusedElement(!!isDirty);`

updates the focused element's dirty state based on the form's dirty state. When `ElementEditor` is rendered again on re-focus after discarding changes, the form is actually still dirty, since we don't reset it when discarding changes. If we `console.log(isDirty)` here, we will realise on first render it's setting the dirty state of `focusedElement` to `true`, then `false` on second render.

Reply:
I see, thanks for spotting this, yes using `reset` is more appropriate here, trouble you to add this of any discarding of changes